### PR TITLE
chore: update cli integ tests to use sdk v3

### PR DESCRIPTION
**Pull request by TheRealAmazonKendra**
### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by TheRealAmazonKendra** [comment]
Sent to the testing pipeline. Will leave listed as WIP until I've verified that succeeds.

**Comment by TheRealAmazonKendra** [comment]
This has succeeded in the test pipeline a few times but I've pushed it with the rebase once again just to cover all my bases.

**Comment by TheRealAmazonKendra** [comment]
Ooops. Then rebased it incorrectly.

**Comment by TheRealAmazonKendra** [comment]
Passed the pipeline again.

File: packages/@aws-cdk-testing/cli-integ/lib/staging/codeartifact.ts [hunk 1]
```diff
@@ -93,27 +114,74 @@ export class TestRepository {
...
+          domain: this.domain,
+          repository: this.repositoryName,
+        }),
+      );
 
       // eslint-disable-next-line no-console
       console.log('Deleted', this.repositoryName);
     } catch (e: any) {
-      if (e.code !== 'ResourceNotFoundException') { throw e; }
+      if (e.code !== 'ResourceNotFoundException') {
```
Comment (scanlonp) [inline packages/@aws-cdk-testing/cli-integ/lib/staging/codeartifact.ts:182]:
```suggestion
      if (e.name !== 'ResourceNotFoundException') {
```

File: packages/@aws-cdk-testing/cli-integ/lib/staging/codeartifact.ts [hunk 2]
```diff
@@ -169,58 +245,77 @@ export class TestRepository {
...
   }
 
   private async domainExists() {
     try {
-      await this.codeArtifact.describeDomain({ domain: this.domain }).promise();
+      await this.codeArtifact.send(new DescribeDomainCommand({ domain: this.domain }));
       return true;
     } catch (e: any) {
-      if (e.code !== 'ResourceNotFoundException') { throw e; }
+      if (e.code !== 'ResourceNotFoundException') {
```
Comment (scanlonp) [inline packages/@aws-cdk-testing/cli-integ/lib/staging/codeartifact.ts:290]:
```suggestion
      if (e.name !== 'ResourceNotFoundException') {
```

File: packages/@aws-cdk-testing/cli-integ/lib/staging/codeartifact.ts [hunk 3]
```diff
@@ -169,58 +245,77 @@ export class TestRepository {
...
   }
 
   private async repositoryExists(name: string) {
     try {
-      await this.codeArtifact.describeRepository({ domain: this.domain, repository: name }).promise();
+      await this.codeArtifact.send(new DescribeRepositoryCommand({ domain: this.domain, repository: name }));
       return true;
     } catch (e: any) {
-      if (e.code !== 'ResourceNotFoundException') { throw e; }
+      if (e.code !== 'ResourceNotFoundException') {
```
Comment (scanlonp) [inline packages/@aws-cdk-testing/cli-integ/lib/staging/codeartifact.ts:302]:
```suggestion
      if (e.name !== 'ResourceNotFoundException') {
```

File: packages/@aws-cdk-testing/cli-integ/lib/staging/codeartifact.ts [hunk 4]
```diff
@@ -264,8 +365,12 @@ async function retryThrottled<A>(block: () => Promise<A>) {
     } catch (e: any) {
       // eslint-disable-next-line no-console
       console.debug(e.message);
-      if (e.code !== 'ThrottlingException') { throw e; }
-      if (attempts-- === 0) { throw e; }
+      if (e.code !== 'ThrottlingException') {
```
Comment (scanlonp) [inline packages/@aws-cdk-testing/cli-integ/lib/staging/codeartifact.ts:368]:
```suggestion
      if (e.name !== 'ThrottlingException') {
```

**Review by scanlonp — CHANGES_REQUESTED** [review]
Committing these changes.

**Review by scanlonp — APPROVED** [review]

**Pull request merged**